services-examples/src/main/java/io/scalecube/services/examples/BootstrapExample.java

```diff
@@ -27,6 +33,9 @@ public static void main(String[] args) throws Exception {
     System.out.println("Start gateway");
     Microservices gateway =
         Microservices.builder()
+            .discovery(
+                (serviceRegistry, serviceEndpoint) ->
+                    new ScalecubeServiceDiscovery(serviceRegistry, serviceEndpoint).start())
```

Comment:
here we invoke `start` and in other places of this class we don't do it, can we get rid of this invoking?

Reply:
let's apply your idea with  a supplier -> when microservices start it will invoke `start` itself

Reply:
> here we invoke start and in other places of this class we don't do it, can we get rid of this invoking?

I will move `start()`  to `ServiceDiscovery`